tf2_geometry_msgs/include/tf2_geometry_msgs/tf2_geometry_msgs.hpp

```diff
@@ -347,6 +349,105 @@ template <>
 inline
   std::array<std::array<double, 6>, 6> getCovarianceMatrix(const geometry_msgs::msg::PoseWithCovarianceStamped& t)  {return covarianceRowMajorToNested(t.pose.covariance);}
 
+/** \brief Transform the covariance matrix of a PoseWithCovariance message to a new frame.
+ * \param cov_in The covariance matrix to transform.
+ * \param transform The transform to apply, as a tf2::Transform structure.
+ * \return The transformed covariance matrix.
+ */
+inline
+geometry_msgs::msg::PoseWithCovariance::_covariance_type transformCovariance(
+  const geometry_msgs::msg::PoseWithCovariance::_covariance_type & cov_in,
+  const tf2::Transform & transform)
+{
+    /**
+     * To transform a covariance matrix:
+     * 
+     * \verbatim[R 0] COVARIANCE [R' 0 ]
+      [0 R]            [0  R']\endverbatim
+     * 
+     * Where:
+     * 	R is the rotation matrix (3x3).
+     * 	R' is the transpose of the rotation matrix.
+     * 	COVARIANCE is the 6x6 covariance matrix to be transformed.
+     *
+     * Reference:
+     * 	A. L. Garcia, “Linear Transformations of Random Vectors,” in Probability,
+     * 	Statistics, and Random Processes For Electrical Engineering, 3rd ed.,
+     * 	Pearson Prentice Hall, 2008, pp. 320–322.
+     */
+
+    // get rotation matrix (and transpose)
+    const tf2::Matrix3x3 R = transform.getBasis();
+    const tf2::Matrix3x3 R_transpose = R.transpose();
+
+    // convert covariance matrix into four 3x3 blocks
+    const tf2::Matrix3x3 cov_11(cov_in[0], cov_in[1], cov_in[2],
+			         cov_in[6], cov_in[7], cov_in[8],
+			         cov_in[12], cov_in[13], cov_in[14]);
```

Comment:
It looks like there are leading tabs in these two lines.  It is causing the function parameters to not line up (both in the GitHub UI and in `vim`).  Please convert all tabs to spaces.

The same goes several times below.

Reply:
Done, thanks to `:retab`.